kolibri/content/utils/check_schema_db.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from sqlalchemy import inspect
+from sqlalchemy.ext.declarative.clsregistry import _ModuleMarker
+from sqlalchemy.orm import RelationshipProperty
+
+logger = logging.getLogger(__name__)
+
+# Modified from https://gist.github.com/miohtama/278fd4eeb9e5272d061c
+
+def db_matches_schema(Base, session):
+    """
+    Check whether the current database matches the models declared in model base.
+    Currently we check that all tables exist with all columns.
+    :param Base: Declarative Base for SQLAlchemy models to check
+    :param session: SQLAlchemy session bound to an engine
+    :return: True if all declared models have corresponding tables and columns.
+    """
+
+    engine = session.get_bind()
+    iengine = inspect(engine)
+
+    errors = False
+
+    tables = iengine.get_table_names()
+
+    # Go through all SQLAlchemy models
+    for name, klass in Base._decl_class_registry.items():
+
+        if isinstance(klass, _ModuleMarker):
+            # Not a model
+            continue
+
+        table = klass.__table__.name
+        if table in tables:
+
+            columns = [c["name"] for c in iengine.get_columns(table)]
+            mapper = inspect(klass)
+
+            for column_prop in mapper.attrs:
+                if isinstance(column_prop, RelationshipProperty):
+                    # TODO: Add sanity checks for relations
+                    pass
+                else:
+                    for column in column_prop.columns:
```

Comment:
Is there any way we can reduce the # of for loops here? Can we get by if we just do a very quick check?

Reply:
Added breaks on error, as one error is as good as a million.